### PR TITLE
Add hosting plan step

### PR DESCRIPTION
**Pull request by EricJizbaMSFT**
For function app advanced creation scenario - user can select dedicated or consumption plan

Related https://github.com/Microsoft/vscode-azurefunctions/issues/854

### CODE REVIEW

File: appservice/src/createAppService/AppServicePlanSkuStep.ts [hunk 1]
```diff
@@ -9,42 +9,48 @@ import { IAzureQuickPickItem } from 'vscode-azureextensionui';
...
 import { IAppServiceWizardContext } from './IAppServiceWizardContext';
 
 export class AppServicePlanSkuStep extends AzureWizardPromptStep<IAppServiceWizardContext> {
     public async prompt(wizardContext: IAppServiceWizardContext): Promise<void> {
-        let pricingTiers: IAzureQuickPickItem<SkuDescription>[] = this.getPlanSkus().map((s: SkuDescription) => {
+        const skus: SkuDescription[] = this.getCommonSkus();
+        if (wizardContext.newSiteOS === WebsiteOS.windows) {
+            skus.unshift(...this.getFreeSkus());
+            if (wizardContext.newSiteKind === AppKind.functionapp) {
+                skus.push(...this.getElasticPremiumSkus());
```
Comment (nturinski) [inline appservice/src/createAppService/AppServicePlanSkuStep.ts]: Get EP plans only available on Windows?

File: appservice/src/createAppService/SiteHostingPlanStep.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep, IAzureQuickPickItem, IWizardOptions } from 'vscode-azureextensionui';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import { AppServicePlanListStep } from './AppServicePlanListStep';
+import { IAppServiceWizardContext } from './IAppServiceWizardContext';
+
+export class SiteHostingPlanStep extends AzureWizardPromptStep<IAppServiceWizardContext> {
```
Comment (nturinski) [inline appservice/src/createAppService/SiteHostingPlanStep.ts]: Where is this going to be used?  I don't see it being added to the wizard anyway.

Comment (nturinski) [reply appservice/src/createAppService/SiteHostingPlanStep.ts]: Never mind, saw your other PR.

File: appservice/src/createAppService/AppServicePlanSkuStep.ts (hunk 1 above)
Comment (EricJizbaMSFT) [reply appservice/src/createAppService/AppServicePlanSkuStep.ts]: Yes

**Pull request merged**